src/gui/application.cpp

```diff
@@ -562,75 +534,6 @@ void Application::slotParseMessage(const QString &msg, QObject *)
-            setHelp();
-            break;
-        } else if (option == QLatin1String("--showsettings") || option == QLatin1String("-s")) {
-            _showSettings = true;
-        } else if (option == QLatin1String("--quit") || option == QLatin1String("-q")) {
-            _quitInstance = true;
-        } else if (option == QLatin1String("--logwindow") || option == QLatin1String("-l")) {
-            _showLogWindow = true;
-        } else if (option == QLatin1String("--logfile")) {
-            if (it.hasNext() && !it.peekNext().startsWith(QLatin1String("--"))) {
-                _logFile = it.next();
-            } else {
-                showHint("Log file not specified");
-            }
-        } else if (option == QLatin1String("--logdir")) {
-            if (it.hasNext() && !it.peekNext().startsWith(QLatin1String("--"))) {
-                _logDir = it.next();
-            } else {
-                showHint("Log dir not specified");
-            }
-        } else if (option == QLatin1String("--logexpire")) {
-            if (it.hasNext() && !it.peekNext().startsWith(QLatin1String("--"))) {
-                _logExpire = std::chrono::hours(it.next().toInt());
-            } else {
-                showHint("Log expiration not specified");
-            }
-        } else if (option == QLatin1String("--logflush")) {
-            _logFlush = true;
-        } else if (option == QLatin1String("--logdebug")) {
-            _logDebug = true;
-        } else if (option == QLatin1String("--confdir")) {
-            if (it.hasNext() && !it.peekNext().startsWith(QLatin1String("--"))) {
-                QString confDir = it.next();
-                if (!ConfigFile::setConfDir(confDir)) {
-                    showHint("Invalid path passed to --confdir");
-                }
-            } else {
-                showHint("Path for confdir not specified");
-            }
-        } else if (option == QLatin1String("--debug")) {
-            _logDebug = true;
-            _debugMode = true;
-        } else if (option == QLatin1String("--version")) {
-            _versionOnly = true;
-        } else if (option == QLatin1String("--language")) {
-            if (it.hasNext() && !it.peekNext().startsWith(QLatin1String("--"))) {
-                _userEnforcedLanguage = it.next();
-            } else {
-                showHint("--language expects a parameter");
-            }
-        } else if (option.endsWith(QStringLiteral(APPLICATION_DOTVIRTUALFILE_SUFFIX))) {
-            // virtual file, open it after the Folder were created (if the app is not terminated)
-            QTimer::singleShot(0, this, [this, option] { openVirtualFile(option); });
-        } else {
-            showHint(QStringLiteral("Unrecognized option '%1'").arg(option));
-        }
-    }
-}
 
 // Helpers for displaying messages. Note that there is probably no console on Windows.
 static void displayHelpText(const QString &t, std::ostream &stream = std::cout)
```

Comment:
Directly call parser.showHelp()

Reply:
I intentionally didn't use this because of that comment. Do you want to change the UX on Windows as well?

Reply:
Recent changes allow us to print to the console on Windows.
Also QCommandLineParser shows the popup itself afaik.

Reply:
We're using `QCommandLineParser` directly now where applicable. As we can't use it to show our own hints, I had to keep `displayHelpText`, though. But `showHint` could be removed.